sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/EmptyNullableValueAccess.cs

```diff
@@ -116,6 +116,50 @@ public void TestComplexCondition(int? i3, double? d3, float? f3)
                 Console.WriteLine(f3.Value); // TODO: Should be NC
             }
         }
+
+        public int CSharp8_SwitchExpressions(bool zero)
+        {
+            int? i = zero switch { true => 0, _ => null };
+            return i.Value; // Noncompliant
+        }
+
+        public int CSharp8_SwitchExpressions2(bool zero)
+        {
+            int? i = zero switch { true => 0, _ => 1 };
+            return i.Value;
+        }
+
+        public int CSharp8_SwitchExpressions3(int? value)
+        {
+            return value.HasValue switch { true => value.Value, false => 0};
+        }
+
+        public int CSharp8_SwitchExpressions4(int? value)
+        {
+            return value.HasValue switch { true => 0, false => value.Value };   // FN - switch expressions are not constrained
```

Comment:
Would be great to add an issue for this (and a comment with the link) to be sure that we don't forget about it.

Reply:
I've created https://github.com/SonarSource/sonar-dotnet/issues/2949 without backlink, there are more rules affected.